### PR TITLE
chore: refactor profile page and account settings to ButtonV2

### PR DESCRIPTION
**Pull request by kopancek**
### Describe what this PR does

First iteration of moving to ButtonV2 in our components, more to come...

This one tackles ProfileButton, Profile Page and Account Settings pages.

Related: https://dailydotdev.atlassian.net/browse/WT-2017 

### Screenshots

Screenshot 2023-12-13 at 10 36 38
<br/>
Screenshot 2023-12-13 at 11 42 40

Screenshot 2023-12-13 at 10 28 26

Screenshot 2023-12-13 at 11 35 59

Screenshot 2023-12-13 at 11 34 58
Screenshot 2023-12-13 at 14 53 18

### On those affected packages:
- [x] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [x] Does this not break anything in companion?

### Did you test the modified components media queries?
- [x] MobileL (420px)
- [x] Tablet (656px)
- [x] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [x] Android

### CODE REVIEW

File: packages/shared/src/components/profile/AccountDangerZone.tsx [hunk 1]
```diff
@@ -46,8 +51,10 @@ function AccountDangerZone({
       {children}
       <Button
         onClick={onDelete}
-        buttonSize={ButtonSize.Small}
-        className="self-start mt-6 btn-primary-ketchup text-theme-label-primary"
+        size={ButtonSize.Small}
+        variant={ButtonVariant.Primary}
+        color={ButtonColor.Ketchup}
+        className="self-start mt-6 text-theme-label-primary"
```
Comment (nensidosari) [inline packages/shared/src/components/profile/AccountDangerZone.tsx:57]: Maybe I missed something like this discussed in buttonV2, or something like this was out of scope, so just checking. 
For `text-theme-label-primary` wouldn't we think a button with a certain color/variant combination has a set label theme color?
Could be something that we can improve later on, but I believe it would be great if a default color for a certain combination is already set and consumers don't have to define that all the time.

No need to take action on this on this PR just raising for discussion and further work improvements we can do on the button.

File: packages/webapp/components/layouts/AccountLayout/Security/index.tsx [hunk 2]
```diff
@@ -209,16 +210,17 @@ function AccountSecurityDefault({
...
+            className="mt-6 w-fit"
             onClick={() => onSwitchDisplay(Display.ChangeEmail)}
           >
             Change email
           </Button>
         )}
       </AccountContentSection>
       <AccountLoginSection
-        className={{ button: 'btn-primary' }}
+        variant={ButtonVariant.Primary}
```
Comment (nensidosari) [inline packages/webapp/components/layouts/AccountLayout/Security/index.tsx:223]: Would this be best as buttonVariant?

**Review by nensidosari — APPROVED** [review]
Looks great, just one comment & suggestion on my side in general idea, but definitely non blocking.

File: packages/shared/src/components/profile/AccountDangerZone.tsx (hunk 1 above)
Comment (kopancek) [reply packages/shared/src/components/profile/AccountDangerZone.tsx:57]: Yes, that's already the case. In this case, it was overriden to white font, so I kept the same visuals as we had before. But agree, ideally we do not add such a classname and keep the default label color.

File: packages/webapp/components/layouts/AccountLayout/Security/index.tsx (hunk 2 above)
Comment (kopancek) [reply packages/webapp/components/layouts/AccountLayout/Security/index.tsx:223]: Good point, renaming :)

File: packages/shared/src/components/profile/AccountDangerZone.tsx (hunk 1 above)
Comment (rebelchris) [reply packages/shared/src/components/profile/AccountDangerZone.tsx:57]: Ah that's actually a good catch.
In the beginning the red color was always with white text we overruled that for accesibility concerns.
So technically you shouldn't have custom colors on buttons.

But maybe double check with design and give them a pre-after screenshot?

File: packages/shared/src/components/profile/ProfileButton.tsx [hunk 3]
```diff
@@ -14,40 +14,38 @@ const ProfileMenu = dynamic(
...
         <SimpleTooltip placement="left" content="Profile settings">
-          <button
-            type="button"
+          <Button
+            variant={ButtonVariant.Float}
+            size={ButtonSize.Small}
             className={classNames(
-              'items-center p-0 ml-0.5 font-bold no-underline rounded-lg border-none cursor-pointer text-theme-label-primary bg-theme-bg-secondary typo-callout focus-outline',
-              className ?? 'flex',
+              'items-center !p-0 ml-0.5 text-theme-label-primary gap-2',
```
Comment (rebelchris) [inline packages/shared/src/components/profile/ProfileButton.tsx:41]: This might be an exception because it's so custom. what's the default color we add on float buttons?

File: packages/shared/src/components/profile/ProfileButton.tsx [hunk 4]
```diff
@@ -14,40 +14,38 @@ const ProfileMenu = dynamic(
...
+              className,
             )}
             onClick={wrapHandler(() => onUpdate(!isOpen))}
           >
-            <span className="hidden laptop:block mr-2 ml-3">
-              {user.reputation ?? 0}
-            </span>
+            <span className="block ml-3">{user.reputation ?? 0}</span>
```
Comment (rebelchris) [inline packages/shared/src/components/profile/ProfileButton.tsx:46]: Why did you remove the responsive styles here?

File: packages/shared/src/components/profile/ProfileButton.tsx [hunk 5]
```diff
@@ -14,40 +14,38 @@ const ProfileMenu = dynamic(
...
-          className="btn btn-tertiary"
+          variant={ButtonVariant.Tertiary}
           onClick={wrapHandler(() => onUpdate(!isOpen))}
           icon={<SettingsIcon />}
         />
       ) : (
         <SimpleTooltip placement="left" content="Profile settings">
-          <button
-            type="button"
+          <Button
```
Comment (rebelchris) [inline packages/shared/src/components/profile/ProfileButton.tsx:37]: After thinking, maybe this shouldn't even be a button tbh 😅
It's like a custom element with a onClick really.
But open to keep it for simplicity

File: packages/webapp/components/layouts/AccountLayout/EmailForm.tsx [hunk 6]
```diff
@@ -55,7 +59,11 @@ function EmailForm({
           label={passwordProps?.label || 'Password'}
         />
       )}
-      <Button className="mt-6 ml-auto w-fit bg-theme-color-cabbage btn-quaternary">
+      <Button
+        className="mt-6 ml-auto w-fit text-theme-label-primary"
```
Comment (rebelchris) [inline packages/webapp/components/layouts/AccountLayout/EmailForm.tsx:63]: I don't think the text override is needed, but maybe also tripple check with design they ok with it.

**Review by rebelchris — CHANGES_REQUESTED** [review]
Minor comments, but overall great work.

You might find a couple places where we in existing places override text colors, imho this shouldn't happen and it's bad practice, we can probably just stick with button guidelines, but in doubt double-check with design team.

File: packages/shared/src/components/profile/ProfileButton.tsx (hunk 3 above)
Comment (kopancek) [reply packages/shared/src/components/profile/ProfileButton.tsx:41]: The default is the same as the bell icon on the notifications button. So it would be greyish - #a8b3cf.

File: packages/webapp/components/layouts/AccountLayout/EmailForm.tsx (hunk 6 above)
Comment (kopancek) [reply packages/webapp/components/layouts/AccountLayout/EmailForm.tsx:63]: I checked and the result is we should keep the default: https://dailydotdev.slack.com/archives/C0381BQUEN9/p1702475962350259?thread_ts=1702475717.398979&cid=C0381BQUEN9

Comment (rebelchris) [reply packages/webapp/components/layouts/AccountLayout/EmailForm.tsx:63]: Thanks for checking, I thought so :)

File: packages/shared/src/components/profile/ProfileButton.tsx (hunk 3 above)
Comment (kopancek) [reply packages/shared/src/components/profile/ProfileButton.tsx:41]: Here is another response from design, we should keep it white: https://dailydotdev.slack.com/archives/C0381BQUEN9/p1702476773654429?thread_ts=1702475717.398979&cid=C0381BQUEN9

As for keeping it as a button or not, I think keeping the Button has better accessibility than having a `<span with onClick>`, e.g. for screen readers it's much easier if it's a `<button` if it's a clickable element. So my vote is for keeping it as a button.

Comment (rebelchris) [reply packages/shared/src/components/profile/ProfileButton.tsx:41]: Ah yes but I mean using a `button` rather then our `Button` component?
I do think this is where `none` variant should actually be used almost.

Comment (rebelchris) [reply packages/shared/src/components/profile/ProfileButton.tsx:41]: Let's do that, it avoids issues down the line if we make buttonv2 even more strict etc.

**Review by rebelchris — APPROVED** [review]
Looks good :)
